{
  final RuntimeException runtimeException=new RuntimeException();
  Reference<Object> reference=FinalizeManager.register(new Object(),new FinalizeAction(){
    @Override public void doFinalize(    Reference<?> reference){
      Assert.assertNotNull(getReferent(reference));
      throw runtimeException;
    }
  }
,FinalizeManager.PHANTOM_REFERENCE_FACTORY);
  Assert.assertNotNull(getReferent(reference));
  reference.enqueue();
  try {
    ReflectionTestUtil.invoke(FinalizeManager.class,"_pollingCleanup",new Class<?>[0]);
    Assert.fail();
  }
 catch (  Exception e) {
    Assert.assertSame(runtimeException,e.getCause());
  }
  Assert.assertNull(getReferent(reference));
}
